vllm/model_executor/models/llama.py

```diff
@@ -276,7 +277,10 @@ def __init__(
                 nn.Identity()
                 for _ in range(self.end_layer, config.num_hidden_layers)
             ])
-        self.norm = RMSNorm(config.hidden_size, eps=config.rms_norm_eps)
+        if get_pp_group().is_last_rank:
+            self.norm = RMSNorm(config.hidden_size, eps=config.rms_norm_eps)
+        else:
+            self.norm = None
```

Comment:
I suppose `norm` should take only a small fraction of GPU memory. is it worth the effort?